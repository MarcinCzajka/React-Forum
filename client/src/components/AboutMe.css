.cardMiddle {
    margin-left: calc(50% - 145px) !important;
    margin-top: 2em !important;
}