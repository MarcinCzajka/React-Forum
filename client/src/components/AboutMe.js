import React from 'react';
import basePath from '../api/basePath';
<<<<<<< HEAD
import { Comment, Form, Button } from "semantic-ui-react";
import moment from "moment";
import UserContext from '../contexts/UserContext';

class AboutMe extends React.Component {
    constructor(props) {
        super(props);

        this.state = {};
    }

    render() {
        return (
            <div style={{display:this.props.display}}>
                me
            </div>
=======
import { Card, Icon, Image } from 'semantic-ui-react';
import moment from 'moment';
import { UserConsumer } from '../contexts/UserContext';

class AboutMe extends React.Component {

    render() {
        return (
            <UserConsumer>
                {context => (
                    <Card style={{display:this.props.display}}>
                        <Image src={context.userAvatar} wrapped ui={false} />
                        <Card.Content>
                        <Card.Header>{context.userName}</Card.Header>
                        <Card.Meta>
                            <span className='date'>Created at: {context.userCreatedAt}</span>
                            <span>{context.userEmail}</span>
                        </Card.Meta>
                        <Card.Description>
                            {context.userDescription}
                        </Card.Description>
                        </Card.Content>
                    </Card>
                )}
            </UserConsumer>
>>>>>>> 6f37e51039f0081358bb8158f0880b973784fe1c
        )
    }

}

export default AboutMe