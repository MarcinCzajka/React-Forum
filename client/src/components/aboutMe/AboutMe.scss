.cardMiddle {
    margin-left: calc(50% - 145px) !important;
    margin-top: 4em !important;
}

.darkFont {
    color: #111 !important;
}

.avatarOverlay {
    position: absolute;
    width: 100%;
    height: 20em;
    opacity: 0;
    background-color: #111;
    cursor: pointer;
    z-index: 5;
    
    transition: all 0.3s ease;
}

.avatarOverlay > i {
    display: block !important;
    position: absolute;
    top: 50%;
    margin-top: -0.5em !important;
    left: 50%;
    margin-left: -0.5em !important;
}

.avatarOverlay:hover {
    opacity: 0.7;
}