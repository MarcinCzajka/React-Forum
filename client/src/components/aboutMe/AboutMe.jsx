import React from 'react';
import { Card, Image, Icon } from 'semantic-ui-react';
import UserContext from '../../contexts/UserContext';
import AvatarPlaceholder from '../placeholders/AvatarPlaceholder';
import moment from 'moment';
import { Helmet } from "react-helmet";
import basePath from '../../api/basePath';
import getCloudinaryWidget from '../../utilities/getCloudinaryWidget';
import './AboutMe.scss';

class AboutMe extends React.Component {
    constructor(props) {
        super(props);

        this.state = {avatarReady: false, avatarFromState: false};
    }

    static contextType = UserContext;

    componentDidMount() {
        getCloudinaryWidget(this.createWidget);
    }
    
    createWidget = () => {
        this.cloudinaryWidget = window.cloudinary.createUploadWidget({
            cloudName: 'dswujhql5', uploadPreset: 'ot93kwr6',
            multiple: false,
            sources: [ 'local', 'url', 'camera' ],
            maxFileSize: 5000000 }, (error, result) => { 
              if (!error && result && result.event === "success") { 

                this.uploadImage(result.info.secure_url);
                this.cloudinaryWidget.close();
              }
            }
        )
    }

    showWidget = () => {
        this.cloudinaryWidget.open();
    }

    onAvatarLoad = () => {
		this.setState({avatarReady: true});
    }
    
    uploadImage = (url) => {
        basePath({
            method: 'put',
            url: `/api/users/avatar/${this.context.userId}`,
            data: {
                avatar: url
            },
            withCredentials: true
        }).then(res => {
            this.setState({avatarFromState: res.data});
        });
    }

    render() {
        const { userName, userAvatar, userEmail, userCreatedAt} = this.context;
        const createdAt = moment(userCreatedAt).format('MMMM Do YYYY, dddd')

        return (
            <Card className='cardMiddle' >
                <Helmet>
                    <title>About Me - React-forum</title>
                </Helmet>

                <div onClick={this.showWidget} className='avatarOverlay'>
                    <Icon name='file outline image' size='huge' inverted></Icon>
                </div>

                {!this.state.avatarReady ? (
                    <AvatarPlaceholder size='sizeAboutMe' /> 
                ) : ''}
                    
                <Image 
                    src={this.state.avatarFromState || userAvatar}
                    className={!this.state.avatarReady ? 'hidden' : ''}
                    onLoad={this.onAvatarLoad} wrapped ui={false}
                />

                <Card.Content>
                <Card.Header>{userName}</Card.Header>
                <Card.Meta>
                    <p className='darkFont'>Created at: {createdAt}</p>
                    <p className='darkFont'>Email: {userEmail}</p>
                </Card.Meta>
                </Card.Content>
            </Card>
        )
    }

}

export default AboutMe