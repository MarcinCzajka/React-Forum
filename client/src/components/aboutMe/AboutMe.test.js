import React from 'react';
import { create } from "react-test-renderer";
import AboutMe from './AboutMe';
import { UserProvider } from '../../contexts/UserContext';

jest.mock('semantic-ui-react/dist/commonjs/addons/Portal/Portal', () => ({ children }) => children);

describe('AboutMe component', () => {
    test('match snapshot /w user logged in 1/2', () => {
        const contextValue = {
            userName: 'User1',
            userEmail: 'Email@g.com',
            userCreatedAt: '2019-12-04T23:04:26.178Z'
        };

        const component = create(<UserProvider value={contextValue}><AboutMe /></UserProvider>);
        expect(component).toMatchSnapshot();
    });

    test('match snapshot /w user logged in 2/2', () => {
        const contextValue = {
            userName: 'U s ER',
            userEmail: 'differentemail@gfds.com',
            userCreatedAt: '2020-02-04T11:32:00.178Z'
        };

        const component = create(<UserProvider value={contextValue}><AboutMe /></UserProvider>);
        expect(component).toMatchSnapshot();
    });

});