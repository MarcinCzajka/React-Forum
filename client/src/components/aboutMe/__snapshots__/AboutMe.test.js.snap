// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutMe component match snapshot /w user logged in 1/2 1`] = `
<div
  className="ui card cardMiddle"
  onClick={[Function]}
>
  <div
    className="avatarOverlay"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="file outline image huge inverted icon"
      onClick={[Function]}
    />
  </div>
  <div
    className="avatarPlaceholderContainer sizeAboutMe"
  >
    <i
      aria-hidden="true"
      className="user icon avatarPlaceholder"
      onClick={[Function]}
    />
  </div>
  <div
    className="image hidden"
    onLoad={[Function]}
  >
    <img />
  </div>
  <div
    className="content"
  >
    <div
      className="header"
    >
      User1
    </div>
    <div
      className="meta"
    >
      <p
        className="darkFont"
      >
        Created at: 
        December 5th 2019, Thursday
      </p>
      <p
        className="darkFont"
      >
        Email: 
        Email@g.com
      </p>
    </div>
  </div>
</div>
`;

exports[`AboutMe component match snapshot /w user logged in 2/2 1`] = `
<div
  className="ui card cardMiddle"
  onClick={[Function]}
>
  <div
    className="avatarOverlay"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="file outline image huge inverted icon"
      onClick={[Function]}
    />
  </div>
  <div
    className="avatarPlaceholderContainer sizeAboutMe"
  >
    <i
      aria-hidden="true"
      className="user icon avatarPlaceholder"
      onClick={[Function]}
    />
  </div>
  <div
    className="image hidden"
    onLoad={[Function]}
  >
    <img />
  </div>
  <div
    className="content"
  >
    <div
      className="header"
    >
      U s ER
    </div>
    <div
      className="meta"
    >
      <p
        className="darkFont"
      >
        Created at: 
        February 4th 2020, Tuesday
      </p>
      <p
        className="darkFont"
      >
        Email: 
        differentemail@gfds.com
      </p>
    </div>
  </div>
</div>
`;
